src/pip/_vendor/chardet/cli/chardetect.py

```diff
@@ -43,7 +43,8 @@ def description_of(lines, name='stdin'):
     u.close()
     result = u.result
     if PY2:
-        name = name.decode(sys.getfilesystemencoding(), 'ignore')
+        fs_enc = sys.getfilesystemencoding() or 'utf-8'
+        name = name.decode(fs_enc, 'ignore')
     if result['encoding']:
         return '{0}: {1} with confidence {2}'.format(name, result['encoding'],
                                                      result['confidence'])
```

Comment:
We would normally expect this change to be reported to the chardet project, and vendor a fixed version when it's released.

Reply:
I have filed a PR for the chardet project here: https://github.com/chardet/chardet/pull/146

Reply:
Reverted in https://github.com/pypa/pip/pull/4972/commits/7534dad76edb904432f3e61502ac3838ad5cd1b2